pkg/api/validation/validation.go

```diff
@@ -2696,32 +2696,47 @@ func validateResourceQuotaScopes(resourceQuota *api.ResourceQuota) field.ErrorLi
 func ValidateResourceQuota(resourceQuota *api.ResourceQuota) field.ErrorList {
 	allErrs := ValidateObjectMeta(&resourceQuota.ObjectMeta, true, ValidateResourceQuotaName, field.NewPath("metadata"))
 
-	fldPath := field.NewPath("spec", "hard")
-	for k, v := range resourceQuota.Spec.Hard {
+	allErrs = append(allErrs, ValidateResourceQuotaSpec(resourceQuota.Spec, field.NewPath("spec"))...)
+	allErrs = append(allErrs, ValidateResourceQuotaStatus(resourceQuota.Status, field.NewPath("status"))...)
+
+	return allErrs
+}
+
+func ValidateResourceQuotaStatus(status api.ResourceQuotaStatus, fld *field.Path) field.ErrorList {
```

Comment:
I may have missed others when I took a pass quickly this afternoon.  Will
look in morning, if there is no clear convention I am not really
opinionated here.

On Wednesday, May 25, 2016, David Eads <notifications@github.com> wrote:

> In pkg/api/validation/validation.go
> <https://github.com/kubernetes/kubernetes/pull/26276#discussion_r64671115>
> :
>
> > @@ -2696,32 +2696,47 @@ func validateResourceQuotaScopes(resourceQuota *api.ResourceQuota) field.ErrorLi
> >  func ValidateResourceQuota(resourceQuota *api.ResourceQuota) field.ErrorList {
> >  	allErrs := ValidateObjectMeta(&resourceQuota.ObjectMeta, true, ValidateResourceQuotaName, field.NewPath("metadata"))
> >
> > -	fldPath := field.NewPath("spec", "hard")
> > -	for k, v := range resourceQuota.Spec.Hard {
> > +	allErrs = append(allErrs, ValidateResourceQuotaSpec(resourceQuota.Spec, field.NewPath("spec"))...)
> > +	allErrs = append(allErrs, ValidateResourceQuotaStatus(resourceQuota.Status, field.NewPath("status"))...)
> > +
> > +	return allErrs
> > +}
> > +
> > +func ValidateResourceQuotaStatus(status api.ResourceQuotaStatus, fld *field.Path) field.ErrorList {
>
> The same convention was followed for the ValidatePod and ValidatePodSpec.
> I guess why not pointers?
>
> Mutation potential? Doesn't indicate what the API is actually trying to do?
>
> I don't have any real object, just seems odd and there some that aren't. I
> don't mind per-se if you think its more correct.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26276/files/77c1219806f9c21421a09386225e8d312dc7df95#r64671115>
>